### PR TITLE
JNG-5834 transform principal

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5834" title="JNG-5834" target="_blank">Sub-taskJNG-5834</a>  Transform Principal
  </td></table>
  <br />

### CODE REVIEW

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/actorDeclaration.eol [hunk 1]
```diff
@@ -58,6 +58,17 @@ operation JSL!ActorGroupDeclaration getAllMenuDeclarations(): Set {
     return menuDeclarations;
 }
 
+@cached
+operation JSL!ActorDeclaration getIdentityTransferDeclaration() : JSL!TransferDeclaration {
+    if (self.getIdentity().isDefined()) {
+        var identityFiled = self.getIdentity().isDefined() ? self.getIdentity().field : null;
+        if (identityFiled.isDefined() and identityFiled.isKindOf(JSL!TransferFieldDeclaration) and identityFiled.eContainer.isKindOf(JSL!TransferDeclaration)) {
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/operations/jsl/data/actorDeclaration.eol]: `if (self.getIdentity().isDefined())` then `self.getIdentity().isDefined() ? ...` ?

File: judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiApplicationTest.java [hunk 2]
```diff
@@ -403,5 +403,22 @@ void testSecurity() throws Exception {
...
+        assertEquals("Actor", actor.getSimpleName());
+
+        // Principal
+
+        ClassType principal = apps.get(0).getPrincipal();
+
+        assertNotNull(principal);
+        assertEquals("SecurityTestModel::UserTransfer::ClassType", principal.getName());
+        assertEquals("UserTransfer", principal.getSimpleName());
+        assertEquals(true, principal.isIsPrincipal());
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiApplicationTest.java]: assertTrue?

File: judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiPagesTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            actor AppActor human;
+        """));
+
+        transform();
+
+        List<Application> apps = uiModelWrapper.getStreamOfUiApplication().toList();
+
+        assertEquals(1, apps.size());
+
+        Application app1 = apps.get(0);
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiPagesTest.java:59]: unused var or is it in progress?

File: judo-tatami-jsl-jsl2ui/src/test/resources/TestShop.jsl [hunk 4]
```diff
@@ -5,104 +5,103 @@ import judo::types;
...
+
+transfer UserTransfer(User user) {
+	field String email <= user.email bind;
 }
 
-actor human CustomerActor(Customer customer) {
-    menu CartView myCart <= customer label:"My cart";
-    menu ProductListView products label:"Products";
+actor CustomerActor human identity:UserTransfer::email {
+//	link CartView myCart <=  label:"My cart";
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/test/resources/TestShop.jsl:105]: Why was this commented?

**Review by bencegelei — APPROVED** [review]

File: judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiPagesTest.java (hunk 3 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiPagesTest.java:59]: The whole process will be "in progress", we will remove unused stuff later if it's okay.

File: judo-tatami-jsl-jsl2ui/src/test/resources/TestShop.jsl (hunk 4 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/test/resources/TestShop.jsl:105]: Ask @gprivi . Model is not used yet anyway.

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**